From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ismael Luceno <ismael@iodev.co.uk>
Date: Mon, 6 Jan 2025 16:28:06 +0100
Subject: [PATCH] gda-connection: Fix pointer type mismatch

Causes a build error with GCC 14, which enables the
-Werror=incompatible-pointer-types flag by default.

Upstream-Status: Pending
Signed-off-by: Ismael Luceno <ismael@sourcemage.org>
---
 libgda/gda-connection.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgda/gda-connection.c b/libgda/gda-connection.c
index 43911e96addb..746204e50175 100644
--- a/libgda/gda-connection.c
+++ b/libgda/gda-connection.c
@@ -6458,7 +6458,7 @@ gda_connection_add_prepared_statement (GdaConnection *cnc, GdaStatement *gda_stm
 
 	if (!cnc->priv->prepared_stmts)
 		cnc->priv->prepared_stmts = g_hash_table_new_full (g_direct_hash, g_direct_equal,
-								   NULL, _gda_prepared_estatement_free);
+								   NULL, (GDestroyNotify)_gda_prepared_estatement_free);
 	g_hash_table_remove (cnc->priv->prepared_stmts, gda_stmt);
 	PreparedStatementRef *ref = _gda_prepared_estatement_new (gda_stmt, prepared_stmt);
 	g_hash_table_insert (cnc->priv->prepared_stmts, gda_stmt, ref);
